taichi/runtime/llvm/runtime.cpp

```diff
@@ -1585,6 +1585,29 @@ DEFINE_SET_PARTIAL_BITS(8);
 DEFINE_SET_PARTIAL_BITS(16);
 DEFINE_SET_PARTIAL_BITS(32);
 DEFINE_SET_PARTIAL_BITS(64);
+
+f32 rounding_prepare_f32(f32 f) {
+  /* slower (but clearer) version with branching:
+  if (f > 0)
+    return f + 0.5;
+  else
+    return f - 0.5;
+  */
+
+  // Branch-free implementation: copy the sign bit of "f" to "0.5"
+  i32 delta_bits =
+      (taichi_union_cast<i32>(f) & 0x80000000) | taichi_union_cast<i32>(0.5f);
+  f32 delta = taichi_union_cast<f32>(delta_bits);
+  return f + delta;
```

Comment:
It's cool to eliminate branch in this way! Now, I am thinking about is it possible and beneficial to implement this part in `Codegen`? Maybe we can discuss this later.

Reply:
That's a good point. I'm just being lazy here. There are two benefits of the current implementation:
- Easy to write
- Easy for other backend authors to learn what's happening here

The cons are
- Need to instantiate for `f32`/`f64` separately
- Slower compilation

We can keep the discussion to our future meetings if you are interested in this :-)